config/acme/machines/config_batch.xml

```diff
@@ -454,8 +461,10 @@
     <job name="case.st_archive">
       <template>template.st_archive</template>
       <task_count>1</task_count>
-      <!-- If DOUT_S is true and case.run (or case.test) exits successfully then run st_archive-->
-      <dependency>case.run or case.test</dependency>
+      <!-- If DOUT_S is true and case.run amd/or case.test exits (regardless of success) then run st_archive
+	   This dependence is 'and' if both are run before case.st, and is 'or' if one is not run
+      -->
+      <dependency>case.run case.test</dependency>
```

Comment:
The logic for this was done in the cime python and not by the underlying batch system.  What indication did you have that this logic was ignored?   I don't think that you should implement this change.